### ISSUE TITLE
Cajoler 500 error in code attempting to use ranges

### ISSUE THREAD

**Issue by kpreid** [google/caja#1521]
Original [issue 1521](https://code.google.com/p/google-caja/issues/detail?id=1521) created by shel.davis@cloudsherpas.com on 2012-09-11T16:00:02.000Z:

<b>What revision of the cajoler exhibits the problem?  On what browser and OS?</b>

Caja Playground
Google Caja. Copyright (C) 2011, Google Inc. Rev 5002 built on 2012-08-02 14:58:24.
(http://caja.appspot.com/)

Chrome 21.0.1180.89
Windows 7, Home Premium

To do this test, I used RangyInputs... RangyInputs is a small cross-browser JavaScript library for obtaining and manipulating selections within &lt;textarea&gt; and &lt;input type=&quot;text&quot;&gt; HTML elements. (http://code.google.com/p/rangyinputs/)

<b>What steps will reproduce the problem?</b>
1. Copy the attached HTML into the Caja playground.
2. Cajole it.
3. See the server error 500.

<b>What is the expected output? What do you see instead?</b>

I expect to see a page like the RangyInputs demo page, where I can type into a TextArea and manipulate it. (http://rangyinputs.googlecode.com/svn/trunk/demos/textinputs_jquery.html)

Instead, I see a server error 500.

<b>Please provide any additional information below.</b>

I've added RangyInputs into Apps Script applications as well, where I can see more feed back from the server. I get the following message, which I assume you'll see in your logs as well.

TextInputs module for Rangy not supported in your browser.
Reason: No means of finding text input caret position 

So, why is the caret position (or even &quot;selected text&quot;) being blocked by Caja?

Thanks in advance for your feedback on this issue.

++++++++++++

&lt;html&gt;
&lt;head&gt;
    &lt;title&gt;Rangyinputs jQuery plugin demo&lt;/title&gt;
    &lt;script type=&quot;text/javascript&quot; src=&quot;http://ajax.googleapis.com/ajax/libs/jquery/1.7.2/jquery.min.js&quot;&gt;&lt;/script&gt;
    &lt;script type=&quot;text/javascript&quot; src=&quot;http://rangyinputs.googlecode.com/files/textinputs_jquery.js&quot;&gt;&lt;/script&gt;
    &lt;style type=&quot;text/css&quot;&gt;
        p.inputcontainer {
            background-color: lightgoldenrodyellow;
            border: solid 1px darkblue;
            padding: 0.3125em;
        }
    &lt;/style&gt;
&lt;/head&gt;
&lt;body&gt;
&lt;h1&gt;Rangy Text Inputs jQuery Demo&lt;/h1&gt;

&lt;br&gt;&lt;br&gt;

&lt;p class=&quot;inputcontainer&quot;&gt;
    &lt;input type=&quot;button&quot; class=&quot;unselectable&quot; value=&quot;get selection range&quot;
           onmousedown=&quot;var sel = $('#ta').getSelection(); alert(sel.start + ',' + sel.end);&quot;&gt;

```
&lt;input type=&quot;button&quot; class=&quot;unselectable&quot; value=&quot;paste test text&quot;
       onmousedown=&quot;$('#ta').replaceSelectedText('**Test**')&quot;&gt;

&lt;input type=&quot;button&quot; class=&quot;unselectable&quot; value=&quot;extract selected text&quot;
       onmousedown=&quot;alert($('#ta').extractSelectedText())&quot;&gt;

&lt;input type=&quot;button&quot; class=&quot;unselectable&quot; value=&quot;surround selected text&quot;
       onmousedown=&quot;$('#ta').surroundSelectedText('[Before]', '[After]', true); $('#ta').focus()&quot;&gt;

&lt;input type=&quot;button&quot; class=&quot;unselectable&quot; value=&quot;replace selected text&quot;
       onmousedown=&quot;$('#ta').replaceSelectedText('[Replacement text]'); $('#ta').focus()&quot;&gt;

&lt;input type=&quot;button&quot; class=&quot;unselectable&quot; value=&quot;collapse to start&quot;
       onmousedown=&quot;$('#ta').collapseSelection(true); $('#ta').focus()&quot;&gt;

&lt;input type=&quot;button&quot; class=&quot;unselectable&quot; value=&quot;collapse to end&quot;
       onmousedown=&quot;$('#ta').collapseSelection(false); $('#ta').focus()&quot;&gt;

&lt;input type=&quot;button&quot; class=&quot;unselectable&quot; value=&quot;set selection&quot;
       onmousedown=&quot;$('#ta').setSelection(+$('#ta_start')[0].value, +$('#ta_end')[0].value); $('#ta').focus()&quot;&gt;

Start: &lt;input type=&quot;text&quot; id=&quot;ta_start&quot; size=&quot;3&quot; value=&quot;5&quot;&gt;
End: &lt;input type=&quot;text&quot; id=&quot;ta_end&quot; size=&quot;3&quot; value=&quot;7&quot;&gt;

&lt;br&gt;&lt;br&gt;
&lt;textarea rows=&quot;15&quot; cols=&quot;80&quot; id=&quot;ta&quot;&gt;Some test text
```

That has

Some line breaks, including some trailing ones

&lt;/textarea&gt;
&lt;/p&gt;
&lt;/body&gt;
&lt;/html&gt;
